File: Projects/loop_race/tenkloops.c
/* Timing 10,000 loop iterations for C 
vs.
Other langs such as Python and Java */

#include <stdio.h>
#include <time.h>

int main()
{

  clock_t start, end;
  double cpu_time_used;
  
  int n = 1; 
  int iterations = 10000;

  start = clock();
  while( n <= iterations)
   {
    printf("%d loops completed\n", n);
    n++;
   }
  
   end = clock();
   cpu_time_used = ((double) (end - start)) / CLOCKS_PER_SEC;

   printf("Time taken: %f s", cpu_time_used);
 
   return 0;
}

File: Projects/loop_race/tenkloops.py
from time import perf_counter


its = 10000
n = 1

start = perf_counter()
while n <= its:
    print(f"{its} loops completed\n")
    n = n + 1

finish = perf_counter()
time = round(finish - start, 1)

print(f"Time taken: {time}s")
